addon/lib/metrics.js

```diff
@@ -121,13 +151,32 @@ module.exports = {
     this.updateExperiment(addonId, {features: features});
   },
 
+  onReceiveVariantDefs: function(ev) {
+    if (!store.experimentVariants) {
+      store.experimentVariants = {};
+    }
+
+    const { subject, data } = ev;
+    const dataParsed = variantMaker.parseTests(JSON.parse(data));
+
+    store.experimentVariants[subject] = dataParsed;
+    Events.emit(EVENT_SEND_VARIANTS, {
+      data: JSON.stringify(dataParsed),
+      subject: self.id
+    });
+  },
+
   onExperimentPing: function(ev) {
     const { subject, data } = ev;
     const dataParsed = JSON.parse(data);
 
     // TODO: Map add-on ID (subject) to other pingTypes as necessary
     const pingType = 'testpilottest';
 
+    if (subject in store.experimentVariants) {
```

Comment:
> We didn't specify this data in the testpilottest ping - though that's a good place for it. 

Why is that a good place for it? I feel like Test Pilot should be agnostic to experiments, while the experiments themselves should be able to segment on these values.